### ISSUE TITLE
Dependabot cannot resolve new patch release of previous major version

### ISSUE THREAD

**Issue by driesvints** [dependabot/dependabot-core#3793]
**Package ecosystem**
npm
**Manifest location and content prior to update**
https://github.com/driesvints/driesvints.com/blob/main/package-lock.json
**Updated dependency**
dns-packet from 1.3.1 to 1.3.3
**What you expected to see, versus what you actually saw**
I see a `dns-packet` vulnerability warning on driesvints/driesvints.com and some other repos (see screenshot at bottom) because of CVE-2021-23386. The transitive dependency (multicast-dns) [is targeting the 1.x releases](https://github.com/mafintosh/multicast-dns/blob/8b510f3b85df6ea8f5c6f1d245621330399f9537/package.json#L11). Dependabot indicates that the earliest fix is 5.2.2 which isn't true. The earliest fix for the 1.x major cycle is 1.3.2 ([see the fixing commit here](https://github.com/mafintosh/dns-packet/commit/0d0d593f8df4e2712c43957a6c62e95047f12b2d)). Dependabot should be perfectly capable of bumping the patch release of dns-packet to 1.3.2 (or better: 1.3.3 which also has a bug fix).

It seems to me that Dependabot isn't capable of bumping previous major release cycle's patches?

I tried searching previous issues but couldn't find any about this specifically.
**Images of the diff or a link to the PR, issue or logs**

![Screenshot 2021-05-26 at 11 12 42](https://user-images.githubusercontent.com/594614/119634828-66021280-be13-11eb-99c7-c178dce957d6.jpg)

**Comment by driesvints** [comment]
I now see the actual issue is that the advisory doesn't have the 1.3.2 version in it. Either the advisory should be adjusted or replaced with a new one. I'm not sure who published this advisory or who to contact to make that happen?

**Comment by asciimike** [comment]
I have filed an issue for the advisories team to look into this and get the other version(s) added. Will close when resolved.

**Comment by asciimike** [comment]
This has been updated, closing.

**Comment by driesvints** [comment]
Thanks @asciimike!

**Comment by driesvints** [comment]
@asciimike it seems Dependabot can still not update the dependency even though it's just a patch bump. I've attached the logs below:

<details><summary>Logs</summary>
<p>

```
proxy | time="2021-05-27T10:07:32Z" level=info msg="proxy starting" commit=37b071acda32d69b1be55b63b62b0f6ff8a76459
  proxy | 2021/05/27 10:07:32 Listening (:1080)
updater | 2021-05-27T10:07:32.887027428 [142068146:main:WARN:src/devices/src/legacy/serial.rs:434] Detached the serial input due to peer close/error.
updater | time="2021-05-27T10:07:34Z" level=info msg="guest starting" commit=35b30a48d278328db64726f4f5a0afa904a6f0ca
updater | time="2021-05-27T10:07:34Z" level=info msg="starting job..." fetcher_timeout=5m0s job_id=142068146 updater_timeout=30m0s updater_version=0.149.0-9c8a3ea9f8779c8a96b4a8ffccbe8ec795a801e4
updater | I, [2021-05-27T10:07:36.114848 #8]  INFO -- sentry: ** [Raven] Raven 3.1.2 ready to catch errors
updater | warning: parser/current is loading parser/ruby26, which recognizes
updater | warning: 2.6.7-compliant syntax, but you are running 2.6.6.
updater | warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
updater | INFO <job_142068146> Starting job processing
  proxy | 2021/05/27 10:07:38 [002] GET https://api.github.com:443/repos/driesvints/driesvints.com
  proxy | 2021/05/27 10:07:38 [002] * authenticating github api request
  proxy | 2021/05/27 10:07:39 [002] 200 https://api.github.com:443/repos/driesvints/driesvints.com
  proxy | 2021/05/27 10:07:39 [004] GET https://api.github.com:443/repos/driesvints/driesvints.com/git/refs/heads/main
  proxy | 2021/05/27 10:07:39 [004] * authenticating github api request
  proxy | 2021/05/27 10:07:39 [004] 200 https://api.github.com:443/repos/driesvints/driesvints.com/git/refs/heads/main
  proxy | 2021/05/27 10:07:39 [006] GET https://api.github.com:443/repos/driesvints/driesvints.com/contents/package.json?ref=a789e89a59e416b2f410c35e01eaacd4f1db4e87
  proxy | 2021/05/27 10:07:39 [006] * authenticating github api request
  proxy | 2021/05/27 10:07:39 [006] 200 https://api.github.com:443/repos/driesvints/driesvints.com/contents/package.json?ref=a789e89a59e416b2f410c35e01eaacd4f1db4e87
  proxy | 2021/05/27 10:07:39 [008] GET https://api.github.com:443/repos/driesvints/driesvints.com/contents/?ref=a789e89a59e416b2f410c35e01eaacd4f1db4e87
  proxy | 2021/05/27 10:07:39 [008] * authenticating github api request
  proxy | 2021/05/27 10:07:39 [008] 200 https://api.github.com:443/repos/driesvints/driesvints.com/contents/?ref=a789e89a59e416b2f410c35e01eaacd4f1db4e87
  proxy | 2021/05/27 10:07:39 [010] GET https://api.github.com:443/repos/driesvints/driesvints.com/contents/package-lock.json?ref=a789e89a59e416b2f410c35e01eaacd4f1db4e87
  proxy | 2021/05/27 10:07:39 [010] * authenticating github api request
  proxy | 2021/05/27 10:07:39 [010] 403 https://api.github.com:443/repos/driesvints/driesvints.com/contents/package-lock.json?ref=a789e89a59e416b2f410c35e01eaacd4f1db4e87
  proxy | 2021/05/27 10:07:39 [012] GET https://api.github.com:443/repos/driesvints/driesvints.com/git/blobs/5e965ca00ee6f72da656628f81edd5866106b67f
  proxy | 2021/05/27 10:07:39 [012] * authenticating github api request
  proxy | 2021/05/27 10:07:39 [012] 200 https://api.github.com:443/repos/driesvints/driesvints.com/git/blobs/5e965ca00ee6f72da656628f81edd5866106b67f
updater | INFO <job_142068146> Finished job processing
updater | time="2021-05-27T10:07:39Z" level=info msg="task complete" container_id=job-142068146-file-fetcher exit_code=0 job_id=142068146 step=fetcher
updater | I, [2021-05-27T10:07:40.830052 #7]  INFO -- sentry: ** [Raven] Raven 3.1.2 ready to catch errors
updater | warning: parser/current is loading parser/ruby26, which recognizes
updater | warning: 2.6.7-compliant syntax, but you are running 2.6.6.
updater | warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
updater | INFO <job_142068146> Starting job processing
updater | INFO <job_142068146> Starting update job for driesvints/driesvints.com
updater | INFO <job_142068146> Checking if dns-packet 1.3.1 needs updating
  proxy | 2021/05/27 10:07:45 [016] GET https://registry.npmjs.org:443/dns-packet
  proxy | 2021/05/27 10:07:48 [016] 200 https://registry.npmjs.org:443/dns-packet
  proxy | 2021/05/27 10:07:48 [018] GET https://registry.npmjs.org:443/dns-packet/5.2.4
  proxy | 2021/05/27 10:07:48 [018] 200 https://registry.npmjs.org:443/dns-packet/5.2.4
updater | INFO <job_142068146> Latest version is 5.2.4
updater | INFO <job_142068146> Requirements to unlock update_not_possible
updater | INFO <job_142068146> Requirements update strategy bump_versions
  proxy | 2021/05/27 10:07:49 [020] GET https://registry.npmjs.org:443/dns-packet/1.3.2
  proxy | 2021/05/27 10:07:49 [020] 200 https://registry.npmjs.org:443/dns-packet/1.3.2
updater | INFO <job_142068146> The latest possible version of dns-packet that can be installed is 1.3.1
updater | INFO <job_142068146> The earliest fixed version is 1.3.2.
updater | INFO <job_142068146> Finished job processing
updater | time="2021-05-27T10:07:50Z" level=info msg="task complete" container_id=job-142068146-updater exit_code=0 job_id=142068146 step=updater
```

</job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></job_142068146></p>
</details> 

![Screenshot 2021-05-27 at 12 08 43](https://user-images.githubusercontent.com/594614/119808384-5bfc1480-bee4-11eb-9aee-f026ae42b883.jpg)

I see:

```
updater | INFO <job_142068146> Checking if dns-packet 1.3.1 needs updating
  proxy | 2021/05/27 10:07:45 [016] GET https://registry.npmjs.org:443/dns-packet
  proxy | 2021/05/27 10:07:48 [016] 200 https://registry.npmjs.org:443/dns-packet
  proxy | 2021/05/27 10:07:48 [018] GET https://registry.npmjs.org:443/dns-packet/5.2.4
  proxy | 2021/05/27 10:07:48 [018] 200 https://registry.npmjs.org:443/dns-packet/5.2.4
updater | INFO <job_142068146> Latest version is 5.2.4
```

Which leads back to the original reason of this issue.
